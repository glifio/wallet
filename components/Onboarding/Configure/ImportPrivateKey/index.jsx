import React, { useState } from 'react'
import Filecoin from '@openworklabs/filecoin-wallet-provider'
import { useDispatch } from 'react-redux'
import { useRouter } from 'next/router'
import {
  Box,
  Button,
  OnboardCard,
  Title,
  Text,
  Input,
  StepHeader,
  LoadingScreen
} from '../../../Shared'
import { walletList } from '../../../../store/actions'
import { useWalletProvider } from '../../../../WalletProvider'
import { createWalletProvider } from '../../../../WalletProvider/state'
import reportError from '../../../../utils/reportError'

export default () => {
  const {
    dispatch,
    fetchDefaultWallet,
    setWalletType,
    walletSubproviders: { SingleKeyProvider }
  } = useWalletProvider()
  const [privateKey, setPrivateKey] = useState('')
  const [privateKeyError, setPrivateKeyError] = useState('')
  const dispatchRdx = useDispatch()
  const [loadingNextScreen, setLoadingNextScreen] = useState(false)
  const router = useRouter()

  const instantiateProvider = async () => {
    try {
      setLoadingNextScreen(true)
      setPrivateKey('')
      const provider = new Filecoin(SingleKeyProvider(privateKey), {
        apiAddress: process.env.LOTUS_NODE_JSONRPC
      })
      dispatch(createWalletProvider(provider))
      const wallet = await fetchDefaultWallet(provider)
      dispatchRdx(walletList([wallet]))
      const params = new URLSearchParams(router.query)
      router.push(`/home?${params.toString()}`)
    } catch (err) {
      reportError(18, false, err.message, err.stack)
      setLoadingNextScreen(false)
      setPrivateKeyError(err.message || JSON.stringify(err))
    }
  }

  return (
    <>
      {loadingNextScreen ? (
        <LoadingScreen />
      ) : (
        <Box
          display='flex'
          flexDirection='column'
          alignItems='center'
          justifyContent='center'
        >
          <OnboardCard>
            <StepHeader
              currentStep={1}
              totalSteps={2}
              description='Import'
              glyphAcronym='Pk'
              showStepper={false}
            />
            <Box
              width='auto'
              display='flex'
              flexDirection='column'
              justifyContent='space-between'
              borderColor='core.lightgray'
              m={2}
            >
              <Title mt={3}>Import</Title>
              <Text>Please input your private key below</Text>
              <Input.PrivateKey
                error={privateKeyError}
                setError={setPrivateKeyError}
                value={privateKey}
                onChange={e => setPrivateKey(e.target.value)}
              />
            </Box>
          </OnboardCard>
          <Box
            mt={6}
            display='flex'
            width='100%'
            justifyContent='space-between'
          >
            <Button
              title='Back'
              onClick={() => setWalletType(null)}
              variant='secondary'
              mr={2}
            />
            <Button
              title='Next'
              disabled={!!(privateKey.length === 0 || privateKeyError)}
              onClick={instantiateProvider}
              variant='primary'
              ml={2}
            />
          </Box>
        </Box>
      )}
    </>
  )
}
