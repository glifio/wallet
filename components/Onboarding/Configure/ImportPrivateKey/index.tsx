import React, { FC, useState } from 'react'
import Filecoin, { SECP256K1KeyProvider } from '@glif/filecoin-wallet-provider'
import { useRouter } from 'next/router'
import {
  Box,
  Button,
  OnboardCard,
  Title,
  Text,
  StepHeader,
  LoadingScreen,
  Input,
  useWalletProvider,
  createWalletProvider
} from '@glif/react-components'
import reportError from '../../../../utils/reportError'
import { navigate } from '../../../../utils/urlParams'
import { PAGE } from '../../../../constants'

const InputPrivateKey: FC<{}> = () => {
  const { dispatch, fetchDefaultWallet, setLoginOption } = useWalletProvider()
  const [privateKey, setPrivateKey] = useState('')
  const [privateKeyError, setPrivateKeyError] = useState('')
  const { walletList } = useWalletProvider()
  const [loadingNextScreen, setLoadingNextScreen] = useState(false)
  const router = useRouter()

  const instantiateProvider = async () => {
    try {
      setLoadingNextScreen(true)
      setPrivateKey('')
      const provider = new Filecoin(new SECP256K1KeyProvider(privateKey), {
        apiAddress: process.env.LOTUS_NODE_JSONRPC
      })
      dispatch(createWalletProvider(provider))
      const wallet = await fetchDefaultWallet(provider)
      walletList([wallet])
      navigate(router, { pageUrl: PAGE.WALLET_HOME })
    } catch (err) {
      reportError(18, false, err.message, err.stack)
      setLoadingNextScreen(false)
      setPrivateKeyError(err.message || JSON.stringify(err))
    }
  }

  return (
    <>
      {loadingNextScreen ? (
        <LoadingScreen />
      ) : (
        <Box
          display='flex'
          flexDirection='column'
          alignItems='center'
          justifyContent='center'
        >
          <OnboardCard>
            <StepHeader
              currentStep={1}
              totalSteps={2}
              description='Import'
              glyphAcronym='Pk'
              showStepper={false}
            />
            <Box
              width='auto'
              display='flex'
              flexDirection='column'
              justifyContent='space-between'
              borderColor='core.lightgray'
              m={2}
            >
              <Title mt={3}>Import</Title>
              <Text>Please input your private key below</Text>
              <Input.PrivateKey
                error={privateKeyError}
                setError={setPrivateKeyError}
                value={privateKey}
                onChange={(e) => setPrivateKey(e.target.value)}
              />
            </Box>
          </OnboardCard>
          <Box
            mt={6}
            display='flex'
            width='100%'
            justifyContent='space-between'
          >
            <Button
              title='Back'
              onClick={() => setLoginOption(null)}
              variant='secondary'
              mr={2}
            />
            <Button
              title='Next'
              disabled={!!(privateKey.length === 0 || privateKeyError)}
              onClick={instantiateProvider}
              variant='primary'
              ml={2}
            />
          </Box>
        </Box>
      )}
    </>
  )
}

export default InputPrivateKey
