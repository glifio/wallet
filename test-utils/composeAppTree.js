import React from 'react'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { initializeStore } from '../test-utils/index'
import { initialState } from '../store/states'

const defaultPreset = {}

const composeAppTree = (preset = defaultPreset, options = {}) => {
  const state = {}
}
