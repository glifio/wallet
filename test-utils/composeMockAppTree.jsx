/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import cloneDeep from 'lodash.clonedeep'

import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { NetworkCheck } from '../lib/check-network'
import BalancePoller from '../lib/update-balance'
import { ConverterContext } from '../lib/Converter'
import { theme, ThemeProvider } from '../components/Shared'
import { initializeStore } from './index'
import { initialState } from '../store/states'
import walletProviderReducer, {
  initialState as walletProviderInitialState,
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../WalletProvider/state'
import { WalletProviderContext } from '../WalletProvider'
import createPath from '../utils/createPath'
import { IMPORT_MNEMONIC } from '../constants'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const mockRouter = jest.fn(() => {})
useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouter,
  replace: mockRouter
}))

jest.mock('@openworklabs/filecoin-wallet-provider')
const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

const mockWalletProviderInstance = new WalletProvider()

const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

const composeWalletProviderState = (initialWalletProviderState, preset) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const mockConverterInstance = options.mockConverterInstance || {
    cacheConversionRate: jest.fn(),
    toFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').dividedBy(5)
    }),
    fromFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').multipliedBy(5)
    })
  }

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )
    return (
      <Provider store={store}>
        <ConverterContext.Provider
          value={{
            converter: mockConverterInstance,
            converterError: options.converterError || null
          }}
        >
          <WalletProviderContext.Provider
            value={{
              state: walletProviderState,
              dispatch: walletProviderDispatch,
              fetchDefaultWallet: jest.fn().mockImplementation(),
              setWalletError: jest
                .fn()
                .mockImplementation(errorMessage =>
                  walletProviderDispatch(setError(errorMessage))
                ),
              setWalletType: jest
                .fn()
                .mockImplementation(walletType =>
                  walletProviderDispatch(setWalletType(walletType))
                ),
              setLedgerProvider: jest.fn(),
              connectLedger: jest.fn(),
              resetLedgerState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetLedgerState())
              }),
              resetState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetState())
              })
            }}
          >
            <NetworkCheck
              networkFromRdx={store.getState().network}
              pathname={pathname}
              query={query}
              switchNetwork={jest.fn()}
            />
            <BalancePoller />
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </WalletProviderContext.Provider>
        </ConverterContext.Provider>
      </Provider>
    )
  }

  return { Tree, store, walletProvider: mockWalletProviderInstance }
}
