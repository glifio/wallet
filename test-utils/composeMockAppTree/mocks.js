import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'

import {
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../../WalletProvider/state'
import { SINGLE_KEY, HD_WALLET, LEDGER } from '../../constants'

jest.mock('@openworklabs/filecoin-wallet-provider')

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

export const mockRouterReplace = jest.fn(() => {})
export const mockRouterPush = jest.fn(() => {})
const unix = jest.spyOn(require('dayjs'), 'unix')

unix.mockImplementation(() => {
  return {
    format: fmt => {
      if (fmt === 'MMM DD') return 'Jan 12'
      if (fmt === 'hh:mmA') return '2:30PM'
    }
  }
})

useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouterPush,
  replace: mockRouterReplace
}))

const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

export const mockWalletProviderInstance = new WalletProvider()

export const defaultMockConverterInstance = {
  cacheConversionRate: jest.fn(),
  toFIL: jest.fn().mockImplementation(amount => {
    return new FilecoinNumber(amount, 'fil').dividedBy(5)
  }),
  fromFIL: jest.fn().mockImplementation(amount => {
    return new FilecoinNumber(amount, 'fil').multipliedBy(5)
  })
}

export const createMockWalletProviderContextFuncs = walletProviderDispatch => ({
  fetchDefaultWallet: jest.fn().mockImplementation(() => ({
    balance: new FilecoinNumber('1', 'fil'),
    address: 't1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq',
    path: ''
  })),
  setWalletError: jest
    .fn()
    .mockImplementation(errorMessage =>
      walletProviderDispatch(setError(errorMessage))
    ),
  setWalletType: jest
    .fn()
    .mockImplementation(walletType =>
      walletProviderDispatch(setWalletType(walletType))
    ),
  setLedgerProvider: jest.fn(),
  connectLedger: jest.fn(),
  resetLedgerState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetLedgerState())
  }),
  resetState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetState())
  })
})

class MockSingleKeyProvider {
  type = SINGLE_KEY

  getAccounts = () => jest.fn()

  sign = () => jest.fn()
}

class MockHDWalletProvider {
  type = HD_WALLET

  getAccounts = () => jest.fn()

  sign = () => jest.fn()
}

export const mockWalletSubproviders = {
  SingleKeyProvider: MockSingleKeyProvider,
  HDWalletProvider: MockHDWalletProvider
}
