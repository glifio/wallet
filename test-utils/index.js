import { createStore } from 'redux'
import { initialState } from '../store/states'
import reducer from '../store/reducer'

export const flushPromises = () => new Promise(setImmediate)

export function initializeStore(state = initialState) {
  return createStore(reducer, state)
}

export default initializeStore
