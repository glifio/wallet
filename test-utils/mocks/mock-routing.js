export const mockRouterReplace = jest.fn(() => {})
export const mockRouterPush = jest.fn(() => {})

export const useRouter = jest.fn().mockImplementation(() => ({
  query: 'network=t',
  push: mockRouterPush,
  replace: mockRouterReplace
}))

export default {
  push: mockRouterPush,
  replace: mockRouterReplace
}
