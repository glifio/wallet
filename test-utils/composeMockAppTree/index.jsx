/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import { Provider } from 'react-redux'
import { Converter } from '@openworklabs/filecoin-number'

import { NetworkCheck } from '../../lib/check-network'
import BalancePoller from '../../lib/update-balance'
import { ConverterContext } from '../../lib/Converter'
import { theme, ThemeProvider } from '../../components/Shared'
import { initializeStore } from '..'
import walletProviderReducer, {
  initialState as walletProviderInitialState
} from '../../WalletProvider/state'
import { WasmContext } from '../../lib/WasmLoader'
import { WalletProviderContext } from '../../WalletProvider'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import createMockWalletProviderContextFuncs from './createWalletProviderContextFuncs.js'
import mockWalletSubproviders from '../mocks/mock-wallet-subproviders'
import { presets, composeWalletProviderState } from './composeState'

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )

    const mockWalletProviderContextFuncs = createMockWalletProviderContextFuncs(
      options.walletProviderDispatch || walletProviderDispatch
    )

    return (
      <Provider store={store}>
        <WasmContext.Provider
          value={{
            generateMnemonic: jest
              .fn()
              .mockImplementation(
                () =>
                  'slender spread awkward chicken noise useful thank dentist tip bronze ritual explain version spot collect whisper glow peanut bus local country album punch frown'
              )
          }}
        >
          <ConverterContext.Provider
            value={{
              converter: new Converter(),
              converterError: options.converterError || null
            }}
          >
            <WalletProviderContext.Provider
              value={{
                state: walletProviderState,
                dispatch:
                  options.walletProviderDispatch || walletProviderDispatch,
                walletSubproviders: {
                  ...mockWalletSubproviders
                },
                ...mockWalletProviderContextFuncs
              }}
            >
              <NetworkCheck
                networkFromRdx={store.getState().network}
                pathname={pathname}
                query={query}
                switchNetwork={jest.fn()}
              />
              <BalancePoller />
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            </WalletProviderContext.Provider>
          </ConverterContext.Provider>
        </WasmContext.Provider>
      </Provider>
    )
  }

  return {
    Tree,
    store,
    walletProvider: mockWalletProviderInstance
  }
}
