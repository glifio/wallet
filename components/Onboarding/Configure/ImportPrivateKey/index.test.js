import { cleanup, render, screen, act, fireEvent } from '@testing-library/react'
import composeMockAppTree from '../../../../test-utils/composeMockAppTree'
import { mockRouterPush } from '../../../../test-utils/mocks/mock-routing'
import { flushPromises } from '../../../../test-utils'

import ImportPrivateKey from '.'

describe('Import private key configuration', () => {
  afterEach(() => {
    jest.clearAllMocks()
    cleanup()
  })
  test('it renders correctly', () => {
    const { Tree } = composeMockAppTree('postOnboard')

    const { container } = render(
      <Tree>
        <ImportPrivateKey />
      </Tree>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it sends the user to wallet view, with a wallet in state upon successful config', async () => {
    const mockWalletProviderDispatch = jest.fn()
    const { Tree, store } = composeMockAppTree('preOnboard', {
      walletProviderDispatch: mockWalletProviderDispatch
    })

    const { container } = render(
      <Tree>
        <ImportPrivateKey />
      </Tree>
    )

    await act(async () => {
      fireEvent.change(screen.getByPlaceholderText('Your private key'), {
        target: { value: 'private key string' }
      })
      await flushPromises()

      fireEvent.click(screen.getByText('Next'))
      await flushPromises()
    })
    expect(container.firstChild).toMatchSnapshot()
    expect(mockRouterPush).toHaveBeenCalledWith('/home?network=t')
    expect(mockWalletProviderDispatch.mock.calls[0][0].type).toBe(
      'CREATE_WALLET_PROVIDER'
    )
    expect(store.getState().wallets.length).toBe(1)
  })
})
