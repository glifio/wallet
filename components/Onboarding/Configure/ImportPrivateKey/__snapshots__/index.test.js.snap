// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import private key configuration it renders correctly 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 4px;
  border-width: 1px;
  padding: 16px;
  display: inline-block;
  width: 100%;
  max-width: 560px;
  min-height: 240px;
  overflow: hidden;
}

.c4 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-style: solid;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  min-width: 0;
  box-sizing: border-box;
  border-color: #C4C4C4;
  margin: 8px;
  width: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c9 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c12 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  min-width: 0;
  box-sizing: border-box;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  min-width: 0;
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c17 {
  min-width: 0;
  box-sizing: border-box;
  margin-top: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c18 {
  cursor: pointer;
  border: 1px solid #262626;
  background-color: transparent;
  border-color: #262626;
  color: #262626;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-right: 8px;
  padding: 16px;
  border-width: 1px;
  border-radius: 4px;
}

.c18:hover {
  opacity: 0.8;
}

.c19 {
  cursor: not-allowed;
  border: 1px solid #C4C4C4;
  background-color: #C4C4C4;
  border-color: #C4C4C4;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  padding: 16px;
  border-width: 1px;
  border-radius: 4px;
}

.c19:hover {
  opacity: 1;
}

.c7 {
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.2;
  font-family: RT-Alias-Grotesk;
  margin: 0;
  margin-top: 16px;
}

.c8 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c10 {
  color: #FC6D6F;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  margin: 0;
  margin-top: 16px;
  margin-bottom: 0;
}

.c14 {
  color: #262626;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c16 {
  min-width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding-left: 8px;
  padding-right: 8px;
  border-radius: 1px;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  font-size: 1.125rem;
  text-align: right;
  cursor: text;
  background: #d1ddfa;
  padding-left: 16px;
  padding-right: 16px;
  width: 480px;
  display: inline-block;
  height: 64px;
  border-radius: 4px;
  border: 0;
  border-color: #999999;
}

.c16:hover {
  background: #bacbf7;
  cursor: text;
}

.c16:focus {
  box-shadow: 0;
  outline: 0;
  background: #bacbf7;
}

.c11 {
  display: inline-block;
  width: 100%;
  border-radius: 1px;
}

.c2 {
  list-style: none;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-color: silver;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
    display="inline-block"
    overflow="hidden"
    width="100%"
  >
    <ul
      class="c2"
      display="flex"
      width="100%"
    >
      <li
        class="c3"
        display="flex"
      >
        <div
          class="c4"
          color="#000"
          display="flex"
          size="6"
        >
          <h3
            class="c5"
          >
            Pk
          </h3>
        </div>
      </li>
    </ul>
    <div
      class="c6"
      display="flex"
      width="auto"
    >
      <h2
        class="c7"
        font-family="RT-Alias-Grotesk"
        font-size="4"
        font-weight="400"
      >
        Import
      </h2>
      <p
        class="c8"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        Please input your private key below
      </p>
      <div
        class="c9"
        display="flex"
      >
        <h4
          class="c10"
          color="status.fail.background"
          font-family="RT-Alias-Grotesk"
          font-size="2"
          font-weight="400"
        />
        <div
          class="c11"
        >
          <div
            class="c12"
            display="flex"
          >
            <div
              class="c13"
              display="flex"
            >
              <h4
                class="c14"
                color="core.nearblack"
                font-family="RT-Alias-Grotesk"
                font-size="2"
                font-weight="400"
              >
                Private key
              </h4>
            </div>
            <div
              class="c15"
              display="flex"
            >
              <input
                class="c16"
                display="inline-block"
                height="7"
                placeholder="Your private key"
                type="password"
                value=""
                width="12"
              />
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c17"
    display="flex"
    width="100%"
  >
    <button
      class="c18"
      font-size="3"
      type="button"
    >
      Back
    </button>
    <button
      class="c19"
      disabled=""
      font-size="3"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Import private key configuration it sends the user to wallet view, with a wallet in state upon successful config 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  margin: 0;
  margin-top: 16px;
}

.c1 {
  display: inline-block;
  height: 48px;
  width: 48px;
  background-color: #5E26FF;
}

.c2 {
  display: inline-block;
  height: 24px;
  width: 24px;
  background-color: rgb(26,0,102);
  -webkit-animation-name: bvGDKI;
  animation-name: bvGDKI;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<div
  class="c0"
  display="flex"
>
  <span
    class="c1"
  >
    <span
      class="c2"
    />
  </span>
  <h4
    class="c3"
    font-family="RT-Alias-Grotesk"
    font-size="2"
    font-weight="400"
  >
    Loading...
  </h4>
</div>
`;
